Sources/Draw/GLRenderer.cpp

```diff
@@ -846,18 +914,117 @@ namespace spades {
 			{
 				GLProfiler::Context p(*profiler, "Clear");
 				device->ClearColor(bgCol.x, bgCol.y, bgCol.z, 1.f);
-				device->Clear((IGLDevice::Enum)(IGLDevice::ColorBufferBit | IGLDevice::DepthBufferBit));
+				device->Clear(
+				  (IGLDevice::Enum)(IGLDevice::ColorBufferBit | IGLDevice::DepthBufferBit));
 			}
 
 			device->FrontFace(IGLDevice::CW);
 
-			{
+			bool visiblePlayers[32];
+
+			if (!client::Client::WallhackActive()) {
 				GLProfiler::Context p(*profiler, "Non-mirrored Objects");
-				RenderObjects();
+				RenderObjects(false);
```

Comment:
Don't refer to `Client` directly from the renderer layer. Introduce adequate abstractions in the renderer API.